contents/docs/advanced/proxy.mdx

```diff
@@ -1,109 +1,168 @@
 ---
-title: Deploying a reverse proxy to PostHog Cloud
+title: Deploy a reverse proxy
 sidebar: Docs
 showTitle: true
 ---
 
-import ProxyPathNamesWarning from './_snippets/proxy-path-names-warning.mdx'
+import ProxyCallout from './_snippets/proxy-callout.mdx'
+import ProxyPlatforms from './proxy/_snippets/proxy-platforms'
+import ProxyDecisionTree from './proxy/_snippets/proxy-decision-tree'
 
-A reverse proxy enables you to send events to PostHog Cloud using your own domain.
+<CalloutBox icon="IconWarning" title="Before you start" type="fyi">
 
-This means setting up a service to redirect requests from a subdomain you choose (like `e.yourdomain.com`) to PostHog.
+If you use a self-hosted proxy, PostHog can't help troubleshoot. Use [our managed reverse proxy](/docs/advanced/proxy/managed-reverse-proxy) if you want support.
 
-You then use this subdomain as your `api_host` in the initialization of PostHog instead of `us.i.posthog.com` or `eu.i.posthog.com`.
+</CalloutBox>
 
-## Why do we recommend deploying one?
+A reverse proxy routes events from your domain to PostHog. Instead of sending events directly to PostHog's domain, you send them through your own subdomain like `e.yourdomain.com`.
```

Comment:
```suggestion
A reverse proxy routes events through your domain to PostHog. Instead of sending events directly to PostHog's domain, you send them through your own subdomain like `e.yourdomain.com`.
```